src/shop/anya.py

```diff
@@ -106,61 +135,45 @@ def shop_loop(self):
-                            ),
-                            inp_img=img,
-                            threshold=0.80
-                        )
-                        if g_gloves.valid:
+                for gg_glove in self.glove_templates['gg']:
+                    wanted_gloves = TemplateFinder(True).search_v2(template_image=gg_glove, image=img, threshold=0.80)
+                    for gloves in wanted_gloves:
+                        if gloves.valid:
                             mouse.click(button="right")
-                            self._messenger.send_message("Bought some decent IAS/+2 gloves")
-                            Logger.info("IAS/+2 gloves bought!")
+                            # TODO: Verify purchases by looking for matches in inventory before and after.
+                            self._messenger.send_message("Bought awesome IAS/+3 gloves!")
+                            Logger.info("IAS/+3 gloves bought!")
                             self.gloves_bought += 1
                             time.sleep(1)
 
             # Select Weapons section
-            if self.look_for_trap_claws is True or self.look_for_melee_claws is True:
-                mouse.move(self.sb_x, self.sb_y, randomize=3, delay_factor=[0.6, 0.8])
+            if self.look_for_trap_claws or self.look_for_melee_claws:
+                mouse.move(self.sb_x, self.sb_y, randomize=3, delay_factor=(0.6, 0.8))
                 wait(0.05, 0.1)
                 mouse.press(button="left")
                 wait(0.3, 0.4)
-                # Search for claws
-                claw_pos = []
+                # Search for claws.
                 img = grab().copy()
                 claw_keys = ["CLAW1", "CLAW2", "CLAW3"]
+                claw_pos = []
                 for ck in claw_keys:
-                    template_match = TemplateFinder(True).search(ck, img, roi=self.roi_vendor)
-                    if template_match.valid:
-                        claw_pos.append(template_match.center)
-                # check out each claw
+                    template_matches = TemplateFinder(True).search_v2(
+                        template_name=ck,
+                        image=img,
+                        roi=self.roi_vendor,
+                        threshold=0.88
+                    )
+                    claw_pos.extend([match.center for match in template_matches])
+                # check out each claw.
                 for pos in claw_pos:
                     # cv2.circle(img, pos, 3, (0, 255, 0), 2)
                     x_m, y_m = convert_screen_to_monitor(pos)
-                    mouse.move(x_m, y_m, randomize=3, delay_factor=[0.5, 0.6])
+                    mouse.move(x_m, y_m, randomize=3, delay_factor=(0.5, 0.6))
                     wait(0.5, 0.6)
                     img_stats = grab()
                     trap_score = 0
                     melee_score = 0
+                    # TODO: Preload these templates.
```

Comment:
I didn't go all the way with preloading all the templates because it seemed overkill, but looking at this and Drognan as well as some other request to add a third merchant made me think there would be a good argument to have a merchant base class with all the shared logic.